### PR TITLE
bugbug/*: Update dependencies

### PR DESCRIPTION
**Pull request by marco-c**

### CODE REVIEW

File: nix/requirements_override.nix [hunk 1]
```diff
@@ -116,6 +116,12 @@ in skipOverrides {
     buildInputs = old.buildInputs ++ [ self."Cython" ];
   };
 
+  "bugbug" = self: old: {
+    patchPhase = ''
+      sed -i 's/python-dateutil==2.8.0/python-dateutil/' requirements.txt
```
Comment (La0) [inline nix/requirements_override.nix]: You're patching your own deps ? :smile:

File: src/bugbug/data/bugbug_data/retriever.py [hunk 2]
```diff
@@ -50,7 +50,8 @@ def retrieve_commits(self):
 
         logger.info('mozilla-central cloned')
 
-        repository.download_commits(self.repo_dir)
+        two_years_and_six_months_ago = datetime.utcnow() - relativedelta(years=2, months=6)
```
Comment (La0) [inline src/bugbug/data/bugbug_data/retriever.py]: Is this code update wanted in the dep update ?

**Comment by La0** [comment]
@marco-c Looks like the Taskcluster build failed, you might want to restart it at some point

**Comment by garbas** [comment]
lets also deploy update-dependencies targets for this projects to production so we stop receiving alerts in irc room

File: nix/requirements_override.nix (hunk 1 above)
Comment (marco-c) [reply nix/requirements_override.nix]: Unfortunately there is something else forcing python-dateutil to 2.6.1, so I can't use the required version from bugbug.

File: src/bugbug/data/bugbug_data/retriever.py (hunk 2 above)
Comment (marco-c) [reply src/bugbug/data/bugbug_data/retriever.py]: It's needed after the update because the `download_commits` function changed.

**Pull request merged**